src/dxtbx/format/FormatNexus.py

```diff
@@ -145,6 +145,15 @@ def get_instrument_name(handle) -> str | None:
             name = None
         return nexus.h5str(name)
 
+    @staticmethod
+    def get_source_probe(handle) -> str | None:
+        try:
+            probe = handle["/entry/source/probe"][()]
+            probe = nexus.h5str(probe)
+        except KeyError:
+            probe = None
+        return probe
+
```

Comment:
Although not necessary here if we're going to rely on the instrument name instead to differentiate as discussed, it might be useful to (separately) add `NXsource.probe` to the `NXsource` class in `nxmx.py`:
https://github.com/cctbx/dxtbx/blob/7494dc0332f94c0d4593865f82c7c08a88ae65ea/src/dxtbx/nexus/nxmx.py#L999-L1015